probe/overlay/weave.go

```diff
@@ -134,9 +134,15 @@ func (w *Weave) Tag(r report.Report) (report.Report, error) {
 	}
 
 	// Put information from weave ps on the container nodes
+	const maxPrefixSize = 12
 	for id, node := range r.Container.Nodes {
-		prefix, _ := node.Latest.Lookup(docker.ContainerID)
-		prefix = prefix[:12]
+		prefix, ok := node.Latest.Lookup(docker.ContainerID)
+		if !ok {
```

Comment:
I think that the kubernetes reporter is *breaking* the lookup invariant. As soon as I stop providing `--probe.kubernetes=true` to the probes I don't get any lookup failures.